llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -4486,14 +4494,67 @@ bool AMDGPUInstructionSelector::isDSOffset2Legal(Register Base, int64_t Offset0,
   return KB->signBitIsZero(Base);
 }
 
+// Whether we can infer the operands are non-negative if the result is
+// non-negative.
+bool canInferNonNegativeOperands(MachineInstr *Addr) {
+  return Addr->getOpcode() == TargetOpcode::G_OR ||
+         (Addr->getOpcode() == TargetOpcode::G_PTR_ADD &&
+          Addr->getFlag(MachineInstr::NoUWrap));
+};
```

Comment:
Extraneous ;

Same observation about the function name: should probably be `canInferNoUnsignedWrap`

Reply:
Maybe just `isNoUnsignedWrap`?

Reply:
Sure